tests/Unit/Core/Domain/Currency/ValueObject/CurrencyExchangeRateTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\Domain\Currency\ValueObject;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CurrencyConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\ExchangeRate;
+
+/**
+ * Class CurrencyExchangeRateTest
+ */
+class CurrencyExchangeRateTest extends TestCase
+{
+    /**
+     * @dataProvider getIncorrectExchangeRates
+     */
+    public function testItThrowsAnExceptionOnIncorrectExchangeRate($incorrectExchangeRate)
+    {
+        $this->expectException(CurrencyConstraintException::class);
+        $this->expectExceptionCode(CurrencyConstraintException::INVALID_EXCHANGE_RATE);
+
+        $exchangeRate = new ExchangeRate($incorrectExchangeRate);
+    }
+
+    public function getIncorrectExchangeRates()
+    {
+        return [
```

Comment:
Well ... it seems a big array with not so much data in it ?

Do you have something you dont like about
```
[0],
[-1],
['0'],
['-1'],
['4.294.967.295,000'],
[null],
[[]],
```